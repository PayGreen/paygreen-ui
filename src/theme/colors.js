const mainColors = {
    green: '#33ad73',
    blue: '#00a99d',
    orange: '#e58a28',
    pink: '#ec6a6a',
    purple: '#a26ae1',
    red: '#b54025',
};

const colors = {
    green: {
        main: mainColors.green,
        gradientBase: '#77d1a6',
        gradientShade: '#a4db95',
    },
    blue: {
        main: mainColors.blue,
        gradientBase: '#51c6be',
        gradientShade: '#76bbd8',
    },
    orange: {
        main: mainColors.orange,
        gradientBase: '#fab56b',
        gradientShade: '#faa272',
    },
    pink: {
        main: mainColors.pink,
        gradientBase: '#e98e90',
        gradientShade: '#e98eb9',
    },
    purple: {
        main: mainColors.purple,
        gradientBase: '#b887ee',
        gradientShade: '#dd94ef',
    },
};

export { mainColors, colors };
