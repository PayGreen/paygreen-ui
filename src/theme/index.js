import { ThemeDefault } from './theme.default';
import { ThemeDark } from './theme.dark';

export { ThemeDefault, ThemeDark };
