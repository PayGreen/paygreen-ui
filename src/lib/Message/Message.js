import React from 'react';
import PropTypes from 'prop-types';
import {
    buttonSizeOptions,
    buttonSizeDefault,
    colorTypeOptions,
    colorTypeDefault,
    colorPalletOptions,
    colorThemeOptions,
    colorThemeDefault,
    formStatusOptions,
    formStatusDefault,
    inputWidthOptions,
    inputWidthDefault,
    blockPositionOptions,
    blockPositionDefault,
    spaceOptions,
    spaceDefault,
} from '../../shared/constants';
import { MessageBase } from './style';

const Message = ({ arrowBlock, children, ...rest }) => {
    const arrow =
        arrowBlock === blockPositionOptions.center
            ? blockPositionOptions.topCenter
            : arrowBlock;

    return (
        <MessageBase {...rest} arrowBlock={arrow}>
            {children}
        </MessageBase>
    );
};

Message.propTypes = {
    colorType: PropTypes.oneOf(Object.values(colorTypeOptions)),
    colorPallet: PropTypes.oneOf([
        colorPalletOptions.theme,
        colorPalletOptions.status,
    ]),
    colorTheme: PropTypes.oneOf(Object.values(colorThemeOptions)),
    colorStatus: PropTypes.oneOf(Object.values(formStatusOptions)),
    blockWidth: PropTypes.oneOf(Object.values(inputWidthOptions)),
    fieldSize: PropTypes.oneOf(Object.values(buttonSizeOptions)),
    hasStaticWidth: PropTypes.bool,
    arrowBlock: PropTypes.oneOf(Object.values(blockPositionOptions)),
    marginTop: PropTypes.oneOf(Object.values(spaceOptions)),
    marginBottom: PropTypes.oneOf(Object.values(spaceOptions)),
};

Message.defaultProps = {
    colorType: colorTypeDefault,
    colorPallet: colorPalletOptions.status,
    colorTheme: colorThemeDefault,
    colorStatus: formStatusDefault,
    blockWidth: inputWidthDefault,
    fieldSize: buttonSizeDefault,
    hasStaticWidth: false,
    arrowBlock: blockPositionDefault,
    marginTop: spaceDefault,
    marginBottom: spaceDefault,
};

export default Message;
