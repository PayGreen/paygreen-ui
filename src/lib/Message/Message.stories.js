import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, radios, select, text } from '@storybook/addon-knobs';
import {
    folder,
    buttonSizeOptions,
    buttonSizeDefault,
    formStatusOptions,
    formStatusDefault,
    inputWidthOptions,
    inputWidthDefault,
    arrowBlockOptions,
    arrowBlockDefault,
    iconSizeOptions,
    fontSizeOptions,
    spaceOptions,
    spaceDefault,
} from '../../shared/constants';
import Text from '../Text/Text';
import Message from './Message';
import { CautionIcon } from '../Icon/Icon';

const arrowBlockLabel = 'Arrow block';
const fieldSizeLabel = 'Field size';
const colorStatusLabel = 'Status color';
const blockWidthLabel = 'Block width';

storiesOf(folder.form + 'Message', module)
    .addDecorator(withKnobs)
    .add('Message', () => (
        <Message
            arrowBlock={radios(
                arrowBlockLabel,
                arrowBlockOptions,
                arrowBlockDefault,
            )}
            colorStatus={select(
                colorStatusLabel,
                formStatusOptions,
                formStatusDefault,
            )}
            fieldSize={radios(
                fieldSizeLabel,
                buttonSizeOptions,
                buttonSizeDefault,
            )}
            blockWidth={radios(
                blockWidthLabel,
                inputWidthOptions,
                inputWidthDefault,
            )}
            marginTop={select('Margin top', spaceOptions, spaceDefault)}
            marginBottom={select('Margin bottom', spaceOptions, spaceDefault)}
        >
            <Text textSize={fontSizeOptions.sm}>
                {text(
                    'Message content',
                    'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque eleifend aliquet libero. Morbi ultrices tristique leo vitae porta.',
                )}
            </Text>
        </Message>
    ))
    .add('Message with Icon', () => (
        <Message
            arrowBlock={radios(
                arrowBlockLabel,
                arrowBlockOptions,
                arrowBlockDefault,
            )}
            colorStatus={select(
                colorStatusLabel,
                formStatusOptions,
                formStatusDefault,
            )}
            fieldSize={radios(
                fieldSizeLabel,
                buttonSizeOptions,
                buttonSizeDefault,
            )}
            blockWidth={radios(
                blockWidthLabel,
                inputWidthOptions,
                inputWidthDefault,
            )}
        >
            <CautionIcon
                iconSize={iconSizeOptions.md}
                colorStatus={select(
                    colorStatusLabel,
                    formStatusOptions,
                    formStatusDefault,
                )}
            />

            <Text textSize={fontSizeOptions.sm}>
                {text(
                    'Message content',
                    'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque eleifend aliquet libero. Morbi ultrices tristique leo vitae porta.',
                )}
            </Text>
        </Message>
    ));
