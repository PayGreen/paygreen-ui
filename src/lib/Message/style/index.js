import styled from 'styled-components';
import { TextBase } from '../../Text/style';
import { arrowBlockStyle, arrowSize } from './base';

const MessageBase = styled.div`
    position: relative;
    max-width: ${props => props.theme.form.inputWidth[props.blockWidth]};
    border-radius: ${props => props.theme.radius[props.fieldSize]};
    padding: ${arrowSize};
    display: grid;
    grid-template-rows: auto 1fr;
    grid-template-areas: 'icon text';
    justify-items: center;
    background-color: ${props =>
        props.theme.status[props.colorStatus].light};
    color: ${props => props.theme.status[props.colorStatus].main};

    & > .icon {
        grid-area: icon;
        margin-right: ${arrowSize};

        svg {
            fill: ${props => props.theme.status[props.colorStatus].main};
        }
    }

    & > ${TextBase} {
        color: ${props => props.theme.status[props.colorStatus].main};
    }

    ${props => arrowBlockStyle[props.arrowBlock]};
`;

export { MessageBase };
