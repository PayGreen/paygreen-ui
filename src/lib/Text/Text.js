import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import {
    textHtmlTagOptions,
    textHtmlTagDefault,

    colorTypeOptions,
    colorTypeDefault,
    colorPalletOptions,
    colorThemeOptions,
    colorThemeDefault,
    greyOptions,
    formStatusOptions,
    formStatusDefault,

    radiusOptions,
    fontSizeOptions,
    fontSizeDefault,
    alignOptions,
    alignDefault,
    spaceOptions,
    spaceDefault,
} from '../../shared/constants';
import { TextBase } from './style';

class Text extends PureComponent {
    render() {
        return <TextBase
            as={this.props.textHtmlTag}
            {...this.props}
        >
            {this.props.children}
        </TextBase>;
    }
}

Text.propTypes = {
    textHtmlTag: PropTypes.oneOf(Object.values(textHtmlTagOptions)),

    colorType: PropTypes.oneOf(Object.values(colorTypeOptions)),
    colorPallet: PropTypes.oneOf(Object.values(colorPalletOptions)),
    colorTheme: PropTypes.oneOf(Object.values(colorThemeOptions)),
    colorWab: PropTypes.oneOf(Object.values(greyOptions)),
    colorStatus: PropTypes.oneOf(Object.values(formStatusOptions)),

    hasBackground: PropTypes.bool,
    radiusSize: PropTypes.oneOf(Object.values(radiusOptions)),
    textSize: PropTypes.oneOf(Object.values(fontSizeOptions)),
    textAlign: PropTypes.oneOf(Object.values(alignOptions)),
    underline: PropTypes.bool,

    paddingLateral: PropTypes.oneOf(Object.values(spaceOptions)),
    paddingTop: PropTypes.oneOf(Object.values(spaceOptions)),
    paddingBottom: PropTypes.oneOf(Object.values(spaceOptions)),

    marginLateral: PropTypes.oneOf(Object.values(spaceOptions)),
    marginTop: PropTypes.oneOf(Object.values(spaceOptions)),
    marginBottom: PropTypes.oneOf(Object.values(spaceOptions)),

    marginInternal: PropTypes.oneOf(Object.values(spaceOptions)),
};

Text.defaultProps = {
    textHtmlTag: textHtmlTagDefault,

    colorType: colorTypeDefault,
    colorPallet: colorPalletOptions.wab,
    colorTheme: colorThemeDefault,
    colorWab: greyOptions.grey60,
    colorStatus: formStatusDefault,

    hasBackground: false,
    radiusSize: radiusOptions.none,
    textSize: fontSizeDefault,
    textAlign: alignDefault,
    underline: false,

    paddingLateral: spaceDefault,
    paddingTop: spaceDefault,
    paddingBottom: spaceDefault,

    marginLateral: spaceDefault,
    marginTop: spaceDefault,
    marginBottom: spaceDefault,

    marginInternal: spaceOptions.sm,
};

export default Text;