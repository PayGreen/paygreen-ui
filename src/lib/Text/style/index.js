import styled from 'styled-components';
import { responsiveSpaces } from '../../../shared/spaces';
import { underline } from '../../Title/style/base';
import {
    htmlTagStyle,
    uppercaseStyle,
    color,
    blockBackground,
    minimizeText,
} from './base';

const TextBase = styled.p`
    ${responsiveSpaces('margin')};
    line-height: ${props => props.theme.font.lineHeight.md};
    text-align: ${props => props.align};
    font-size: ${props => props.theme.font.size[props.textSize]};

    ${props => (props.isResponsive ? minimizeText : null)};
    ${props => (props.hasBackground ? blockBackground[props.colorType] : null)};
    ${props => htmlTagStyle[props.htmlTag]};
    ${props => (props.hasUnderline ? underline : null)};
    ${props => color[props.colorType]};
    ${props => (props.hasUppercase ? uppercaseStyle : null)};
`;

export { TextBase };
