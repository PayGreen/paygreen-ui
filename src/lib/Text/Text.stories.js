import React from 'react';
import Text from './Text';
import {
    colorTypeOptions,
    colorTypeDefault,
    colorThemeOptions,
    colorThemeDefault,
    greyOptions,
    transparentColorOptions,
    transparentColorDefault,
    fontSizeOptions,
    fontSizeDefault,
    alignOptions,
    alignDefault,
    textHtmlTagOptions,
    blockPaddingOptions,
    blockPaddingDefault,
    spaceOptions
} from '../../shared/constants';
import { storiesOf } from '@storybook/react';
import { withKnobs, radios, select } from '@storybook/addon-knobs';

storiesOf('Text', module)
    .addDecorator(withKnobs)
    .add('Single paragraph', () => (
        <Text
            colorType={radios('Color type', colorTypeOptions, colorTypeDefault)}
            mainColor={select('Main color', greyOptions, greyOptions.grey60)}
            colorTheme={radios('Color of <strong>', colorThemeOptions, colorThemeDefault)}
            backgroundColor={radios('Background-color', transparentColorOptions, transparentColorDefault)}
            textSize={select('Size', fontSizeOptions, fontSizeDefault)}
            textAlign={radios('Align', alignOptions, alignDefault)}
            paddingBlock={select('Block\'s padding', blockPaddingOptions, blockPaddingDefault)}
            paddingTop={select('Padding top', blockPaddingOptions, blockPaddingDefault)}
            paddingBottom={select('Padding bottom', blockPaddingOptions, blockPaddingDefault)}
            marginTop={select('Block\'s margin top', blockPaddingOptions, blockPaddingDefault)}
            marginBottom={select('Block\'s margin bottom', blockPaddingOptions, blockPaddingDefault)}
        >
            Lorem ipsum dolor sit amet, consectetur <strong>adipiscing elit</strong>. Donec eget <em>nulla in libero</em> laoreet sodales. Fusce vestibulum at neque nec convallis. Vivamus quis neque vulputate, fringilla massa vitae, interdum ipsum. Duis enim augue, euismod quis aliquam in, consequat quis risus.
        </Text>
    ), {
        notes: 'Text component must be used to adding any text without specific component, or inside components like Card.',
    })
    .add('Block of paragraphs', () => (
        <Text
            textHtmlTag={textHtmlTagOptions.div}
            colorType={radios('Color type', colorTypeOptions, colorTypeDefault)}
            mainColor={select('Main color', greyOptions, greyOptions.grey60)}
            colorTheme={radios('Color of <strong>', colorThemeOptions, colorThemeDefault)}
            backgroundColor={radios('Background-color', transparentColorOptions, transparentColorDefault)}
            textSize={select('Size', fontSizeOptions, fontSizeDefault)}
            textAlign={radios('Align', alignOptions, alignDefault)}
            paddingBlock={select('Left and right padding', blockPaddingOptions, blockPaddingDefault)}
            paddingTop={select('Padding top', blockPaddingOptions, blockPaddingDefault)}
            paddingBottom={select('Padding bottom', blockPaddingOptions, blockPaddingDefault)}
            marginTop={select('Block\'s margin top', blockPaddingOptions, blockPaddingDefault)}
            marginBottom={select('Block\'s margin bottom', blockPaddingOptions, blockPaddingDefault)}
            marginInternal={select('Internal margin', spaceOptions, spaceOptions.sm)}
        >
            <p>
                Lorem ipsum dolor sit amet, consectetur <strong>adipiscing elit</strong>. Donec eget nulla in libero laoreet sodales. Fusce vestibulum at neque nec convallis. Vivamus quis neque vulputate, fringilla massa vitae, interdum ipsum. Duis enim augue, euismod quis aliquam in, consequat quis risus.
            </p>

            <p>
                Etiam diam elit, fringilla sit amet blandit sit amet, <em>rutrum varius est</em>. Integer gravida vel urna nec interdum. Suspendisse vel quam et libero pellentesque aliquet. Sed est velit, dignissim eget nunc a, ornare tempor odio. In efficitur, leo in vestibulum posuere, tellus enim porttitor nisl, a tincidunt mauris quam sit amet mi.
            </p>

            <p>
                Nam rhoncus ligula posuere justo consequat. Suspendisse nunc nulla, pellentesque at tellus et, hendrerit facilisis dolor. Sed a dignissim mauris.
            </p>
        </Text>
    ), {
        notes: 'If you want to add more than paragraph, put your text inside <code>&lt;p&gt;</code> tags and pass textHtmlTag prop to "div".',
    })
    .add('Paragraph with background', () => (
        <Text
            colorType={radios('Color type', colorTypeOptions, colorTypeDefault)}
            colorTheme={radios('Color of <strong>', colorThemeOptions, colorThemeDefault)}
            backgroundColor={transparentColorOptions.theme}
            paddingBlock={blockPaddingOptions.lg}
            paddingTop={blockPaddingOptions.lg}
            paddingBottom={blockPaddingOptions.lg}
        >
            Lorem ipsum dolor sit amet, consectetur <strong>adipiscing elit</strong>. Donec eget nulla in libero laoreet sodales. Fusce vestibulum at neque nec convallis. Vivamus quis neque vulputate, fringilla massa vitae, interdum ipsum. Duis enim augue, euismod quis aliquam in, consequat quis risus.
        </Text>
    ), {
        notes: 'You can add background-color, margins and paddings on your Text component.',
    });