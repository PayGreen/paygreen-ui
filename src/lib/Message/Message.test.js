import React from 'react';
import TestRenderer from 'react-test-renderer';
import { ThemeDefault } from '../../theme';
import Text from '../Text/Text';
import Message from './Message';

it('renders without crashing', () => {
    const message = TestRenderer.create(
        <Message theme={ThemeDefault}>
            <Text theme={ThemeDefault}>
                Lorem ipsum dolor sit amet, consectetur non sodales.
            </Text>
        </Message>,
    );
    expect(message.toJSON()).toMatchSnapshot();
});
