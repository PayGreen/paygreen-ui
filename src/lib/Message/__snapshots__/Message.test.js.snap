// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="style__MessageBase-ts6lfl-0 hgewSx"
>
  <p
    className="style__TextBase-sc-11ixwvd-0 dwEpwZ"
  >
    Lorem ipsum dolor sit amet, consectetur non sodales.
  </p>
</div>
`;
